packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts

```diff
@@ -64,3 +64,20 @@ export class DirectiveWrapper {
 export const generateGetArgumentsInput = ({ shouldDeepMergeDirectiveConfigDefaults }: TransformParameters): GetArgumentsOptions => ({
   deepMergeArguments: shouldDeepMergeDirectiveConfigDefaults,
 });
+
+/**
+ * Checks for cases where we don't need to do deep cloning and merging of arguments.
+ * These include cases when the user provided arguments are empty or when there are no common keys between the default and user provided arguments.
+ * @param defaultValue the default properties set for the directive
+ * @param argValues the user provided arguments for the directive
+ * @returns if deep cloning and merging of arguments is needed
+ */
+export const needsDeepMerge = <T>(defaultValue: Required<T>, argValues: { [x: string]: any }): boolean => {
+  if (_.isEmpty(argValues)) {
+    return false;
+  }
+  if (typeof defaultValue === 'object') {
```

Comment:
Won't this be always true?

Reply:
Since `defaultValue` is a generic, this will filter out if any scalars like strings are passed to it.

Reply:
Can we say `needsDeepMerge = <T extends object>(defaultValue: Required<T>, argValues: { [x: string]: any })`?